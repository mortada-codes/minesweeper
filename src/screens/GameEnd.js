
import React,{useEffect,} from 'react';
import { Animated, View,Text,Button,StyleSheet,Dimensions,Easing,} from 'react-native';

const { height } = Dimensions.get('window')




export default  ({onPressRetry,hasWin})=>{

    const scale = new  Animated.Value(1)  
    const translateY = new  Animated.Value(height)
    
    useEffect(()=>{
        Animated.sequence([
            Animated.timing(translateY,{toValue:0,easing:Easing.out(Easing.ease),duration:1000,useNativeDriver:true}),
            Animated.loop(Animated.timing(scale,{toValue:1.4,easing:Easing.out(Easing.ease),duration:100,useNativeDriver:true,}),{iterations:4,resetBeforeIteration:true}),
            Animated.timing(scale,{toValue:1,easing:Easing.out(Easing.ease),duration:100,useNativeDriver:true,})
        ]).start()
    },[])
 
    return (
        <View style={styles.container}>
        <Animated.View style={[styles.content,{transform:[{scaleX:scale},{scaleY:scale},{translateY}]}]} >
              <Text style={styles.text} testID="GameEndMessage">{hasWin?'YOU WON':'GAME OVER'}</Text>  
                <Button onPress={onPressRetry} title="RETRY"/>
        </Animated.View>
        </View>
    )
}


const styles = StyleSheet.create({

  container:{
      justifyContent:'center',
  alignItems:'center',
  height:'100%',
  width:'100%',
  position:'absolute',
    zIndex:1,
    },
    content:{
        backgroundColor:'purple',
        height:200,
        width:'80%',
        justifyContent:'center',
    },
    text:{
        fontSize:22,
        fontWeight:'bold',
        textAlign:'center'      
    }  
})