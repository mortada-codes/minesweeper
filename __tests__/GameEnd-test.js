/**
 * @format
 */

import 'react-native';
import React from 'react';
import GameEnd from '../src/screens/GameEnd';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { fireEvent, render, wait } from '@testing-library/react-native';


 
it('renders correctly', () => {
    global.withAnimatedTimeTravelEnabled(() => { 
        global.timeTravel(1500);  
        const {asJSON} =  render(<GameEnd />);
        expect(asJSON()).toMatchSnapshot();

    })
});


it('Show win message', () => {
    global.withAnimatedTimeTravelEnabled(() => { 
        global.timeTravel(1500);  
        const {getByTestId} =  render(<GameEnd hasWin={true}/>);
        expect(getByTestId("GameEndMessage").props.children).toBe("YOU WON");

    })
});


it('Show lose message', () => {
    global.withAnimatedTimeTravelEnabled(() => { 
        global.timeTravel(1500);  
        const {getByTestId} =  render(<GameEnd hasWin={false}/>);
        expect(getByTestId("GameEndMessage").props.children).toBe("GAME OVER");

    })
});


it('Button title equals retry', () => {
    global.withAnimatedTimeTravelEnabled(() => { 
        global.timeTravel(1500);  
        const {getByTitle} =  render(<GameEnd hasWin={false}/>);
        expect(getByTitle("RETRY"));

    })
});
