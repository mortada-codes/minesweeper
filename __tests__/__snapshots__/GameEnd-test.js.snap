// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "purple",
          "height": 200,
          "justifyContent": "center",
          "transform": Array [
            Object {
              "scaleX": 1,
            },
            Object {
              "scaleY": 1,
            },
            Object {
              "translateY": 1334,
            },
          ],
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 22,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
        testID="GameEndMessage"
      >
        GAME OVER
      </Text>
      <Button
        title="RETRY"
      />
    </View>
  </View>
</View>
`;
